test-util/src/main/java/io/camunda/zeebe/test/util/record/ProcessInstanceRecordStream.java

```diff
@@ -42,6 +42,14 @@ public ProcessInstanceRecordStream withProcessInstanceKey(final long processInst
     return valueFilter(v -> v.getProcessInstanceKey() == processInstanceKey);
   }
 
+  public ProcessInstanceRecordStream withProcessInstanceKeyOrParentProcessInstanceKey(
+      final long processInstanceKey) {
+    return valueFilter(
+        v ->
+            v.getProcessInstanceKey() == processInstanceKey
+                || v.getParentProcessInstanceKey() == processInstanceKey);
+  }
+
```

Comment:
❓ Do we have any other places where we use a similar check that could be refactored to use this new method?

Reply:
Honestly, I didn't look. I can imagine it's a useful filter for other test cases, hence why I put it in here. It's less straightforward to use my IDE to find tests that could use it currently.

I tried some text searching and couldn't find anything, but obviously this method fails as soon as something is written different.